src/main/java/com/redhat/composer/model/mongo/MongoDbSetup.java

```diff
@@ -0,0 +1,39 @@
+package com.redhat.composer.model.mongo;
+
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.IndexOptions;
+import io.quarkus.runtime.StartupEvent;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import org.bson.Document;
+
+/**
+ * Adds unique constraints to fields in the admin collections: llm, retriever and assistants.
+ */
+@ApplicationScoped
+public class MongoDbSetup {
+
+  @Inject
+  MongoClient mongoClient;
+
+  /**
+   * Setup unique constraints on Mongo collections.
+   * @param event the startup event
+   */
+  public void onStart(@Observes StartupEvent event) {
+    MongoDatabase database = mongoClient.getDatabase("composer");
```

Comment:
Could we add this to the liquibase instead of here just to make it consistent?
https://docs.liquibase.com/change-types/add-unique-constraint.html